### PR TITLE
refactor: add more tests

### PR DESCRIPTION
**Pull request by imos**
Claude Code 試してみました。

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+### Documentation
+- Add comprehensive docstrings for new functions
+- Include parameter and return type information
+- Document edge cases and limitations
+- Provide usage examples for complex functions
+
+## Known Issues and Limitations
+
+### Current Technical Debt
```
Comment (masanorihirano) [inline CLAUDE.md:130]: ここら辺、今回のPRで解消されているのであれば治すべきでは？

File: tests/functions/test_cumcount.py [hunk 2]
```diff
@@ -14,3 +14,234 @@ def test_cumcount() -> None:
...
+    result = QF.cumcount(x)
+    expected = torch.tensor([0, 1, 2, 3, 4])
+    np.testing.assert_array_equal(result.numpy(), expected.numpy())
+
+
+def test_cumcount_2d_tensors_dim0() -> None:
+    """Test cumulative counting on 2D tensors along dimension 0."""
+    x = torch.tensor([[1, 2, 3], [1, 2, 4], [1, 3, 3]])
+    result = QF.cumcount(x, dim=0)
+    expected = torch.tensor([[0, 0, 0], [1, 1, 0], [2, 0, 1]])
```
Comment (masanorihirano) [inline tests/functions/test_cumcount.py:47]: ここ、あっている？
torch.tensor([[0, 0, 0], [1, 1, 0], [2, 0, **0**]])
ではない？

File: CLAUDE.md (hunk 1 above)
Comment (imos) [reply CLAUDE.md]: CLAUDE.mdはsimplifyしました。

**Comment by imos** [comment]
Claude Codeのレビュー

⏺ Code Review for PR #4: "refactor: add more tests"

  Overview

  This PR represents a massive improvement to the test suite, adding 18,611 lines of comprehensive tests while removing only 337 lines. It transforms the testing infrastructure from basic coverage to a
  systematic, thorough testing framework.

  Strengths

  1. Test Infrastructure ✅

  - New test_utils.py module with reusable testing utilities
  - Consistent patterns for dtype/device preservation checks
  - Memory efficiency and consistency validators

  2. Comprehensive Coverage ✅

  - 35+ new test files for previously untested functions
  - Systematic test categories: basic, edge cases, dimensions, dtypes, special values
  - Each function tested across multiple scenarios and tensor configurations

  3. Edge Case Handling ✅

  - Empty tensors, single elements, all-NaN/all-infinity tensors
  - Mixed special values (NaN, +inf, -inf)
  - Leading/trailing NaN patterns for fill operations
  - Negative dimension indexing

  4. Documentation ✅

  - Added CLAUDE.md with concise development guidelines
  - Clear testing requirements and patterns
  - Practical instructions for using uv and make commands

  Code Quality

  - Naming Convention: Excellent - test_{function}_{specific_case}() pattern
  - Test Organization: Well-structured with clear test purposes
  - Reference Comparisons: Good use of NumPy/SciPy for validation
  - Type Safety: Proper dtype preservation testing

  Minor Suggestions

  1. Consider pytest parameterization for repetitive dtype tests:
  @pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
  def test_function_dtype(dtype): ...

  2. Test data generators could reduce boilerplate for common patterns
  3. Group related tests in classes for larger test files

  Security & Performance

  - No security issues identified
  - Performance tests included for large tensors
  - Memory efficiency tests prevent leaks

  Recommendation

  APPROVE ✅ - This PR significantly enhances test quality and coverage. The systematic approach and reusable utilities establish excellent testing patterns for future development.

File: tests/functions/test_nankurtosis.py [hunk 3]
```diff
@@ -1,26 +1,69 @@
...
+                20.0,
+            ],  # 10 values
+            [1.1, 2.2, nan, 3.3, 4.4, 5.5, 6.6, 7.7, 8.8, 9.9],  # 9 values
         ],
         dtype=torch.float64,
     )
-    np.testing.assert_allclose(
-        QF.nankurtosis(x, dim=1, unbiased=False).numpy(),
-        scipy.stats.kurtosis(x, axis=1, bias=True, nan_policy="omit"),
-    )
```
Comment (masanorihirano) [inline tests/functions/test_nankurtosis.py:23]: これも消しちゃっていいんでしたっけ？

File: tests/random/test_random.py [hunk 4]
```diff
@@ -7,52 +7,58 @@
 from qfeval_functions import random as qfeval_random
 
 
-class TestSeed:
-    def test_seed(self) -> None:
-        with qfeval_random.seed():
-            a = self.generate_random_values()
-            b = self.generate_random_values()
-        rand_values = []
```
[the remark this answers is not in the document]
Comment (imos) [reply tests/random/test_random.py:15]: これはクラスを解除しているだけなので大丈夫なはずです。
test_seedが一致するものです。

File: tests/functions/test_pca.py [hunk 5]
```diff
@@ -6,35 +6,34 @@
...
-        )
-        # Test if applying PCA to a batch of matrices works.
-        batch_returns = returns[None, None].expand(3, 4, 100, 10)
-        batch_result = QF.pca(batch_returns).components
-        assert batch_result.shape == (3, 4, 10, 10)
-        np.testing.assert_array_almost_equal(
-            np.abs(batch_result.numpy()[0, 0]),
-            np.abs(expected),
-            decimal=3,
-        )
```
[the remark this answers is not in the document]
Comment (imos) [reply tests/functions/test_pca.py:33]: こちらも意図的に解除しています（最初全部クラスにしていたものの名残です）

File: tests/functions/test_nankurtosis.py (hunk 3 above)
Comment (imos) [reply tests/functions/test_nankurtosis.py:23]: xの内容が拡充されてはいますが、基本的にexpectedとresultに先に代入してから確認しているだけなのでtest_nankurtosisで同様のテストが行われていると思います。

**Review by masanorihirano — APPROVED** [review]
